### ISSUE TITLE
Minor issues

### ISSUE THREAD

**Issue by cazala** [hardlydifficult/DecentralandCreeps#1]
Hi @hardlydifficult ! This repo was forwarded to me by @nearnshaw to take a look at it in order to use it for a tutorial/video. First of all, great job! I really like the approach you took to build this game, I just wanted to give you some feedback around some minor issues that I think can be improved:

* The name of the ScriptableScene class is still `HouseScene` ([link](https://github.com/hardlydifficult/DecentralandCreeps/blob/master/server/RemoteScene.tsx#L12))

* The name of the list of enemies/creeps is `entities` (i.e. `getState().entities`), I think this name might be a little misleading since `entity` is a term used to refer to the 3D entities in the scene like `<entity>`, maybe a name like `enemies` or `creeps` or `blobs` would be more clear? Same goes for the `<Entity>` component ([link](https://github.com/hardlydifficult/DecentralandCreeps/blob/master/server/components/Entity.tsx))

* I see this pattern repeated in the code:

```ts
// do something
setTimeout(() => {
  // do something else later
}, 1000)
```

This pattern becomes a little hard to follow when there are many nested timeouts like [here](https://github.com/hardlydifficult/DecentralandCreeps/blob/master/server/RemoteScene.tsx#L87-L124), but if you add a helper like this:

```tsx
const sleep = (ms: number): Promise<void> => new Promise(resolve => setTimeout(resolve, ms))
```

Then you can use a pattern like this:

```ts
// do something
await sleep(100)
// do something else later
await sleep(1000)
// do something else after that
```

Which is easier to follow, you just have to make sure to make the function you are executing that helper in an `async` function, like converting this:

```ts
this.eventSubscriber.on('event', () => {
  // do something
  setTimeout(() => {
    // do something else
  }, 1000)
})
```

into this:

```ts
this.eventSubscriber.on('event', async () => {
  // do something
  await sleep(1000)
  // do something else
})
```

notice the `async`

```diff
- this.eventSubscriber.on('event', () => {
+ this.eventSubscriber.on('event', async () => {
```

* And finally, just as a suggestion (it's probably out of the scope for this tutorial) I see that the game starts as soon as the server starts, but there's no concept of a "match". If this were to be deployed to decentraland it would be a never ending match (unless the server restarts). It would be great if the game had some event/condition to make it start (ie clicking on a button) and then an "end" condition (time/score/lives/etc) to make it end the game and go back to initial state. Again, that's just a suggestion if you are planning to deploy this to DCL (which I believe you should totally do!) but I don't think is necessary in order to use this as a tutorial/video.

**Comment by HardlyDifficult** [comment]
Thanks for the feedback!  I've made all the changes you suggested and will begin crafting the tutorial now.

**Comment by cazala** [comment]
Awesome :D
